Changes

harsh0620 opened the pull request:
<!--
The manual review may take several days or weeks,
depending on the reviewer's availability and workload.
Patience padawan!

You can request a review from @packagecontrol-bot.
Please ensure the reviews pass and follow any instructions.

Please provide some information via this checklist,
feel free to remove what's not applicable.
-->

- [1] I'm the package's author and/or maintainer.
- [1] I have have read [the docs][1].
- [1] My package repo has a description and a README describing what it's for and how to use it.
- [1] Any commands are available via the command palette.

My package is 'Snippet Cpp' which is the extension of snippets for sublime in c++ with the help of which one can do fast coding in Competitive Coding. I have also added all the Algorithms and Ways to tackle Data Structure like Tree, BST, and Graph so that while solving any problem one does not have to google the code if they know the concept. 

<!-- OR -->
My package is similar to https://packagecontrol.io/packages/CPP%20Competitive%20Programming%20Snippets, However, it should still be added because I added more snippets that are necessary for the Competitive Programming as well as fast coding of Data Structure and Algorithm. On top of that, the quoted package has only a single snippet for any Data Structure but I added all the Algorithms and Ways to tackle Data Structure. You can check Graph, Tree, and BST sections.


<!-- 
*)   Unless it definitely really needs them,
     they apply to the cursor's context
     and their visibility is conditional.
     Space in this menu is limited!
**)  There aren't enough keys for all packages,
     you'd risk overriding those of other packages.
     You can put commented-out suggestions in a keymap file, 
     and/or explain how to create bindings in your README.
***) We have hundreds of color schemes,
     and plenty of scopes to make any syntax work. 

For bonus points also consider how the review guidelines apply to your package:
https://github.com/wbond/package_control_channel/wiki#reviewing-a-package-addition

For updates to existing packages:
If your package isn't using tag-based releases,
please switch to tags now.
 -->

[1]: https://packagecontrol.io/docs/submitting_a_package
[2]: https://semver.org
[3]: https://www.git-scm.com/docs/gitattributes#_export_ignore

braver commented:
Please let us know why you're proposing this change.

harsh0620 commented:
I want to create a sublime package and I am following this documentation(https://packagecontrol.io/docs/submitting_a_package) for the same. While following the steps I have to get the pr request to merge, that's why I am proposing this change.

braver commented:
Ah, that's not how you add a package. Add your entry to [repository/s.json](https://github.com/wbond/package_control_channel/blob/master/repository/s.json). And then come back to [complete the checklist](https://github.com/wbond/package_control_channel/blob/master/.github/PULL_REQUEST_TEMPLATE.md) and explain why your package should be added.

harsh0620 commented:
I added the file as per your reply and also updated the checklist. Please have a look and suggest to me any changes that I have to do add this package.

braver commented:
Please don't format the whole file, and please remove your entry from channel.json.

harsh0620 commented:
Removed my entry from the channel.json

braver commented:
> Please don't format the whole file

Your changes cover the whole file. Please just add your package without also formatting everything else.

harsh0620 commented:
Please look if the change is now ok. It is the same as in the main s.json file of this repo but with the addition of my package.

braver commented:
Alright, that's better. Still though, why can't your changes be added to the existing package? That would benefit all current and potential future users of that package.

harsh0620 commented:
So it's the end. I will not be able to add my package on sublime either that package is not yet updated after its publishing in 2017.

braver commented:
Have you tried opening a PR on that existing package? If that doesn't work out we can consider replacing it with yours.

harsh0620 commented:
I tried but there is no revert back from the owner.

harsh0620 commented:
Can you please now guide me to the next steps?

braver commented:
You can try pinging them here in this PR, and then we’ll give them 2 weeks to respond. After that we can go ahead and replace the existing package and bring the improvements to all existing users.

> Op 7 jan. 2022 om 13:43 heeft Harsh Chandravanshi ***@***.***> het volgende geschreven:
> 
> ﻿
> Can you please now guide me to the next steps?
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. 
> You are receiving this because you commented.

harsh0620 commented:
I tried pinging several times for two weeks but there is no reply so now we can go-ahead?

braver commented:
Indeed, the package from @MeghaSharma21 seems well and truly dead. The best way to go now would be to replace the existing package with yours. You can change the repo url in the existing entry (and make any other changes you’d like) instead of adding a new entry.

harsh0620 commented on repository/s.json:
```diff
@@ -5906,4 +5917,4 @@
 		}
 	]
-}
+}
```
Please have a look is the changes enough and at right place.

harsh0620 commented:
Please review the changes

braver commented:
Again, please don’t format the entire file and review your proposed changes closely.
